openvdb/openvdb/cmd/CMakeLists.txt

```diff
@@ -116,8 +117,12 @@ endif()
 #### vdb_render
 
 if(OPENVDB_BUILD_VDB_RENDER)
-  find_package(IlmBase ${MINIMUM_ILMBASE_VERSION} REQUIRED COMPONENTS Half Iex IlmThread Imath)
-  find_package(OpenEXR ${MINIMUM_OPENEXR_VERSION} REQUIRED COMPONENTS IlmImf)
+  find_package(Imath CONFIG)
+  if (NOT TARGET Imath::Imath)
+    find_package(IlmBase ${MINIMUM_ILMBASE_VERSION} REQUIRED COMPONENTS Half Iex IlmThread Imath)
+  endif()
+
+  find_package(OpenEXR ${MINIMUM_OPENEXR_VERSION} REQUIRED)
```

Comment:
Does OpenEXR come with a config modue? Can we not do the same thing here as above (search for OpenEXR CONFIG and recover if the target doesn't exist with our Find scripts)

Reply:
OpenEXR 3 has a config module, but not OpenEXR 2.x. If I'm not mistaken the approach Cary is using here is following the pattern in the OpenEXR 3 migration guide written by @lgritz

Reply:
Yeah so it seems this pattern is being follows for Imath, but not for OpenEXR https://github.com/AcademySoftwareFoundation/Imath/blob/master/docs/PortingGuide2-3.md#openexrimath-3x-only

I'd expect:
```
find_package(Imath CONFIG)
if (NOT TARGET Imath::Imath)
  # use our modules for both ilmbase and openexr
else()
   find_package(OpenEXR CONFIG)
endif()
```

Basically only use our FindOpenEXR and FindIlmBase modules if we can't find CONFIG modules. Assuming the CONFIG modules were introduce in OpenEXR 3 then we don't even need to make any changes to our Find modules right?

Reply:
I just pushed a commit that follows your suggestion. I'm not entirely sure if it's right, but it seems to build properly with 2.x and 3.x in my tests.

Reply:
My thought process here is as follows: We ideally shouldn't have FindOpenEXR and FindIlmBase modules in OpenVDB, we should exclusively use your CONFIG modules. As EXR 3 comes with these, we can first try searching for libs with the CONFIG signature and, if that fails, we can revert back to the MODULE signature. This has the effect of looking for EXR 3 first and falling back to EXR 2.X with our own modules.

We needed these modules for 2 reasons a) EXR didn't install CONFIG files and b) we needed custom location of exr libs from 3rdparty packages such as Houdini. I don't think we need the latter anymore as the EXR symbols provided by Houdini namespaced so we need our own copy anyway (I think this is correct).

Reply:
A complicating factor is that some OpenEXR 2.x releases also had config files, to differing degrees of brokenness, and in any case, the names of the targets of the OpenEXR 2.x config files are different from those of the OpenEXR 3.x config files.

I've found that what you should do is look for `find_package(Imath CONFIG)` first, if that succeeds, it's definitely 3.x you're dealing with (because there was no separate Imath package for 2.x. I think this is what you're leaning toward above as well.

Reply:
I just pushed another commit that fixes the code block in question here to add find_package(OpenEXR) alongside the find_package(IlmBase) in the case where it can't find Imath, i.e. version 2.x, which I believe retains the previous behavior. I'd accidentally removed that, so the build checks failed. Hopefully this gets things back to working order.